core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/upgrade/validation/handler/ClusterUpgradeDiskSpaceValidationHandler.java

```diff
@@ -56,6 +62,12 @@ protected Selectable doAccept(HandlerEvent<ClusterUpgradeValidationEvent> event)
         }
     }
 
+    private StatedImage getTargetImage(ClusterUpgradeValidationEvent request, Long stackId)
+            throws CloudbreakImageNotFoundException, CloudbreakImageCatalogException {
+        Image currentImage = imageService.getImage(stackId);
+        return imageCatalogService.getImage(currentImage.getImageCatalogUrl(), currentImage.getImageCatalogName(), request.getImageId());
+    }
+
```

Comment:
This functionality is present in ImageService, it might make sense to use that:
https://github.com/hortonworks/cloudbreak/blob/28dc0a1cbbebcd7244b30e5de9e2f14b9f48615f/core/src/main/java/com/sequenceiq/cloudbreak/service/image/ImageService.java#L96